src/gui/modalMenu.cpp

```diff
@@ -398,3 +398,15 @@ porting::AndroidDialogState GUIModalMenu::getAndroidUIInputState()
 	return porting::getInputDialogState();
 }
 #endif
+
+GUIModalMenu::scaling_info GUIModalMenu::getScalingInfo(v2u32 base_size, v2u32 screensize)
+{
+	scaling_info info{m_gui_scale, base_size};
```

Comment:
Nitpick: What if it's both too large horizontally and vertically, but the factor is smaller vertically? Then the latter will override the former, and you will still have horizontal cutoff.

I think this can be simplified (and corrected) to `return scaling_info {std::min(m_gui_scale, static_cast<f32>(screensize.X) / static_cast<f32>(base_size.X), static_cast<f32>(screensize.Y) / static_cast<f32>(base_size.Y))), base_size};`

Reply:
> Nitpick: What if it's both too large horizontally and vertically, but the factor is smaller vertically? Then the latter will override the former, and you will still have horizontal cutoff.

That's not how it works, feel free to test.

Reply:
Tested, you are right. It works because it takes the scale factor into account when it does the second check, I overlooked that :facepalm: